pkg/infra/log/log.go

```diff
@@ -433,9 +442,55 @@ func ReadLoggingConfig(modes []string, logsPath string, cfg *ini.File) error {
 		configLoggers = append(configLoggers, handler)
 	}
 
+	var err error
+	root.gokitLogActivated, err = isNewLoggerActivated(cfg)
+	if err != nil {
+		return err
+	}
 	if len(configLoggers) > 0 {
 		root.initialize(configLoggers)
 	}
 
 	return nil
 }
+
+// This would be removed eventually, no need to make a fancy design.
+// For the sake of important cycle I just copied the function
+func isNewLoggerActivated(cfg *ini.File) (bool, error) {
+	section := cfg.Section("feature_toggles")
+	toggles, err := readFeatureTogglesFromInitFile(section)
+	if err != nil {
+		return false, err
+	}
+	return toggles["gokitlogActivated"], nil
+}
+
+func readFeatureTogglesFromInitFile(featureTogglesSection *ini.Section) (map[string]bool, error) {
+	featureToggles := make(map[string]bool, 10)
+
+	// parse the comma separated list in `enable`.
+	featuresTogglesStr := valueAsString(featureTogglesSection, "enable", "")
+	for _, feature := range util.SplitString(featuresTogglesStr) {
+		featureToggles[feature] = true
+	}
+
+	// read all other settings under [feature_toggles]. If a toggle is
+	// present in both the value in `enable` is overridden.
+	for _, v := range featureTogglesSection.Keys() {
+		if v.Name() == "enable" {
+			continue
+		}
+
+		b, err := strconv.ParseBool(v.Value())
+		if err != nil {
+			return featureToggles, err
+		}
+
+		featureToggles[v.Name()] = b
+	}
+	return featureToggles, nil
+}
+
+func valueAsString(section *ini.Section, keyName string, defaultValue string) string {
+	return section.Key(keyName).MustString(defaultValue)
+}
```

Comment:
given that `ReadLoggingConfig()` is called from `github.com/grafana/grafana/pkg/setting` where the original `ReadFeatureTogglesFromInitFile()` is defined,
I think that it would be preferable if you added a new boolean parameter to `ReadLoggingConfig()` whether the new log format is activated or not (in that you way you wouldn't need to re-define them here)

Reply:
That was my first reaction except that we would have an import cycle since log -> setting -> log in that case, this code would be removed quite quickly, I don't think a refractory on settings is really necessary.

Reply:
In what I'm proposing we don't need to import `setting` from `log`. 
I suggest to define `isNewLoggerActivated()` in `setting` and pass it to `ReadLoggingConfig()`.

An alternative would be to define toggles in `Cfg` and read them before the logging initiation but this is a bigger change and I'm not sure we want to export them anyway.